code/ui/manager/src/components/sidebar/Search.stories.tsx

```diff
@@ -20,76 +20,105 @@ const getLastViewed = () =>
     .filter((item, i) => item.type === 'component' && item.parent && i % 20 === 0)
     .map((component) => ({ storyId: component.id, refId }));
 
+let queryParams: Record<string, string> = {};
+
 const meta = {
   component: Search,
   title: 'Sidebar/Search',
   parameters: { layout: 'fullscreen' },
+  args: {
+    dataset,
+    getLastViewed: (): Selection[] => []
+  },
+  render: (args) => {
+    return (
+      <Search {...args}>
+        {() => <SearchResults {...noResults} />}
+      </Search>
+    );
+  },
   decorators: [
-    (storyFn: any) => (
-      <div style={{ padding: 20, maxWidth: '230px' }}>
-        <IconSymbols />
-        {storyFn()}
-      </div>
+    (storyFn) => (
+      <ManagerContext.Provider
+        value={
+          {
+            state: {
+            },
+            api: {
+              emit: () => {},
+              on: () => {},
+              off: () => {},
+              getShortcutKeys: () => ({ search: ['control', 'shift', 's'] }),
+              selectStory: () => {},
+              setQueryParams: (params: Record<string, string>): void => {
+                queryParams = params;
+              },
+              getQueryParams: () => queryParams
```

Comment:
here you could use a mock functions instead of the top-level `queryParams`, and then mock the return value per story and assert on the `setQueryParams` as you want.

```
import { fn } from '@storybook/test';

...

const setQueryParamsMock = fn().mockName('setQueryParams');
const getQueryParamsMock = fn().mockName('getQueryParams');
```

```suggestion
              setQueryParams: setQueryParamsMock,
              getQueryParams: getQueryParamsMock
```

then in tests:

```ts
const MyStory = {
  ...,
  beforeEach: () => {
    getQueryParamsMock.mockReturnValue({ whatever });
  },
  play: () => {
    expect(setQueryParamsMock).toHaveBeenCalledWith({ ... });
  }
}
```

The `beforeEach` hook is being introduced in 8.1 as we speak so it makes sense that you wouldn't know about that yet. https://storybook.js.org/docs/8.1/writing-stories/mocking-modules#using-mocked-modules-in-stories

Reply:
aha i didn't realise we had a `beforeEach`! all good, i'll update 👍